models/user.go

```diff
@@ -38,11 +39,33 @@ type User struct {
 // Validate vaildate all of the user's input
 func (u *User) Validate() error {
 	if u.Username == "" {
-		return errors.New("User username is empty")
+		return errors.New("Username cannot be empty")
+	}
+
+	m, err := regexp.MatchString("^[a-zA-Z0-9@._-]*$", u.Username)
+	if err != nil {
+		return err
```

Comment:
Not sure what's the err.Error() output, but it will be directly exposed the end user, may be good to normalize it returning a generic message like: `"Password can only contain the following characters: a-z 0-9 @._-"`